From ab028918d3b1060d4fce8e17a5ef31315dec10c4 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Fri, 21 Sep 2018 10:32:00 +0530
Subject: [PATCH 0415/1851] drm: xlnx: mixer: Fix linking failure for ARCH arm

commit c656e67621c6c012f742d0e9de48c0be00188f96 from
https://github.com/Xilinx/linux-xlnx.git

The kernel fails to build on 32-bit arm architecture when Video Mixer
driver is enabled due to unsupported modulo operations in 32 bit mode.
The error shown is "undefined reference to `__aeabi_uldivmod'".
This is fixed by changing the data type of luma and chroma address to
dma_addr_t instead of u64.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/xlnx_mixer.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/xlnx/xlnx_mixer.c b/drivers/gpu/drm/xlnx/xlnx_mixer.c
index 1d9a0f97b6e9..a41d2951504b 100644
--- a/drivers/gpu/drm/xlnx/xlnx_mixer.c
+++ b/drivers/gpu/drm/xlnx/xlnx_mixer.c
@@ -1181,7 +1181,8 @@ static int xlnx_mix_disp_set_layer_alpha(struct xlnx_mix_plane *plane,
  */
 static int xlnx_mix_set_layer_buff_addr(struct xlnx_mix_hw *mixer,
 					enum xlnx_mix_layer_id id,
-					u64 luma_addr, u64 chroma_addr)
+					dma_addr_t luma_addr,
+					dma_addr_t chroma_addr)
 {
 	struct xlnx_mix_layer_data *layer_data;
 	u32 align, offset;
@@ -1593,7 +1594,7 @@ static int xlnx_mix_set_plane(struct xlnx_mix_plane *plane,
 	struct xlnx_mix *mixer;
 	struct drm_gem_cma_object *luma_buffer;
 	u32 luma_stride = fb->pitches[0];
-	u64 luma_addr, chroma_addr = 0;
+	dma_addr_t luma_addr, chroma_addr = 0;
 	u32 active_area_width;
 	u32 active_area_height;
 	enum xlnx_mix_layer_id layer_id;
-- 
2.31.1

